integration-tests/steps/change_sets.py

```diff
@@ -0,0 +1,224 @@
+            context.error = e
+            return
+        else:
+            raise e
+    context.output = response
+
+
+@when('the user executes change set "{change_set_name}" for stack "{stack_name}"')
+def step_impl(context, change_set_name, stack_name):
+    environment_name, basename = os.path.split(stack_name)
+    env = Environment(context.sceptre_dir, environment_name)
+    stack = env.stacks[basename]
+    allowed_errors = {'ValidationError', 'ChangeSetNotFound'}
+    try:
+        stack.execute_change_set(change_set_name)
+    except ClientError as e:
+        if e.response['Error']['Code'] in allowed_errors:
+            context.error = e
+            return
+        else:
+            raise e
+
+
+@when('the user describes change set "{change_set_name}" for stack "{stack_name}"')
+def step_impl(context, change_set_name, stack_name):
+    environment_name, basename = os.path.split(stack_name)
+    env = Environment(context.sceptre_dir, environment_name)
+    stack = env.stacks[basename]
+    allowed_errors = {'ValidationError', 'ChangeSetNotFound'}
+    try:
+        response = stack.describe_change_set(change_set_name)
+    except ClientError as e:
+        if e.response['Error']['Code'] in allowed_errors:
+            context.error = e
+            return
+        else:
+            raise e
+    context.output = response
+
+
+@then('stack "{stack_name}" has change set "{change_set_name}" in "{state}" state')
+def step_impl(context, stack_name, change_set_name, state):
+    full_name = get_cloudformation_stack_name(context, stack_name)
+
+    status = get_change_set_status(context, full_name, change_set_name)
+
+    assert status == state
+
+
+@then('stack "{stack_name}" does not have change set "{change_set_name}"')
+def step_impl(context, stack_name, change_set_name):
+    full_name = get_cloudformation_stack_name(context, stack_name)
+    status = get_change_set_status(context, full_name, change_set_name)
+
+    assert status is None
+
+
+@then('the change sets for stack "{stack_name}" are listed')
+def step_impl(context, stack_name):
+    full_name = get_cloudformation_stack_name(context, stack_name)
+    response = retry_boto_call(
+        context.client.list_change_sets,
+        StackName=full_name
+    )
+
+    assert response["Summaries"] == context.output["Summaries"]
+
+
+@then('no change sets for stack "{stack_name}" are listed')
+def step_impl(context, stack_name):
+    full_name = get_cloudformation_stack_name(context, stack_name)
+
+    assert context.output["Summaries"] == []
+
+
+@then('change set "{change_set_name}" for stack "{stack_name}" is described')
+def step_impl(context, change_set_name, stack_name):
+    full_name = get_cloudformation_stack_name(context, stack_name)
+    response = retry_boto_call(
+        context.client.describe_change_set,
+        StackName=full_name,
+        ChangeSetName=change_set_name
+    )
+    response.pop("ResponseMetadata")
+    context.output.pop("ResponseMetadata")
```

Comment:
We pop the `ResponseMetadata` because it will change on every request - as it contains attribute such as `RequestId` which is unique and will never be the same. However the rest of the response should be the same as the response provided by Sceptre. I will change above to be consistent.